### PR TITLE
Travis patchelf pypi_deploy Updates

### LINKED ISSUE #806: auditwheel requires an updated version of patchelf.
*Please refer to our [FAQ](https://ark-analysis.readthedocs.io/en/latest/_rtd/faq.html) and look at our [known issues](https://github.com/angelolab/ark-analysis/issues) before opening a bug report.*

**Describe the bug**

Travis cannot run test_pypi_deploy due to `auditwheel` requiring a newer version of `patchelf`.

**Expected behavior**

`test_pypi_deploy` should run automatically when there is a tag.

**To Reproduce**
Use `xenial` as the distribution, and run the `test_pypi_deploy` stage.

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Closes #806. Fixes an issue with building the wheel for the `test_pypi_deploy` job. 

**How did you implement your changes**

Moved `apt-get install patchelf` to run before creating the `bdist` wheel. Updated the distribution from `xenial` (16.04) to `jammy` (22.04), as the necessary version of patchelf is not available for `xenial`. Added a line in the development doc file stating to modify the travis file to run `test_pypi_deploy` for a new release. In addition needed to pin `multiprocess` until other libraries use the latest version of `dill`.

**Remaining issues**

N/A

### CODE REVIEW

**Review by alex-l-kong — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -29,20 +29,20 @@ jobs:
...
       script:
         - "travis_wait 120 sleep 7200 &"
-        - python setup.py bdist_wheel
         - sudo apt-get install patchelf
+        - python setup.py bdist_wheel
         - auditwheel-symbols --manylinux 2_24 dist/*.whl
         - auditwheel repair dist/*.whl --plat manylinux_2_17_x86_64
         - twine upload -r testpypi --skip-existing /home/travis/build/"$TRAVIS_REPO_SLUG"/wheelhouse/*.whl -u "$TEST_PYPI_USERNAME" -p "$TEST_PYPI_PASSWORD"
     - stage: pypi_deploy
       if: tag IS present
```
Comment (ngreenwald) [inline .travis.yml:41]: Will you comment/then uncomment this line so that we can have a version of the previous build on pypi?

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Looks good, just need to update pypi with previous version that never got sent

File: .travis.yml (hunk 1 above)
Comment (srivarra) [reply .travis.yml:41]: Yeah, should I do the same to the Docker upload as well?

Comment (ngreenwald) [reply .travis.yml:41]: Yup, otherwise anyone who tries to run the current start_docker.sh will get an error

**Comment by ngreenwald** [comment]
So the current version of `start_docker.sh` now works?

**Comment by srivarra** [comment]
@ngreenwald this is good to go.

**Pull request merged**